ui/helpers/utils/i18n-helper.tsx

```diff
@@ -6,20 +6,25 @@ import getFetchWithTimeout from '../../../shared/modules/fetch-with-timeout';
 
 const fetchWithTimeout = getFetchWithTimeout();
 
-const warned = {};
-const missingMessageErrors = {};
-const missingSubstitutionErrors = {};
+const warned = {} as any;
+const missingMessageErrors = {} as any;
+const missingSubstitutionErrors = {} as any;
```

Comment:
We should avoid using any as much as possible while still keeping the goal of decent levels of conversion in our sprint in mind. However its important to note that the more we rely upon typecasting to any the less beneficial the conversion to TypeScript actually is.  I'd like us to reserve the usage of any to places where we truly don't know yet what the types are or if typing it would require a much deeper process. In those cases we should come back shortly after this sprint to make a plan for handling them.

In this case here a a set of types:

```ts
// In our app/_locales folders such as app/_locales/en there is a messages.json file.
// Each file is an object with keys that are translation keys. In our codebase you'll see
// t('reject') the 'reject' is the translationKey. The values of the object are an object with
// this shape. Message is the string that will replace the translationKey, and that message
// may contain replacement variables such as $1, $2, etc. Description is key describing the
// usage of the message. 
interface I18NMessage {
  message: string;
  description?: string;
}

// Now that we have the shape of the individual entries in the messages.json file, this is the
// shape of overall file itself, translationKey (string) and the I18NMessae as the value.
interface I18NMessageDict {
  [translationKey: string]: I18NMessage;
}

// Below we are going to need a couple maps that use the same structure (translationKey) key
// but with different values, so we create a parameterized type (or generic type) that allows us
// to create a shortcut to what we want.
interface I18NMessageDictMap<R> {
  [translationKey: string]: R;
}

// We can then type these objects for safety in the below code.
const warned: { [localeCode: string]: I18NMessageDictMap<boolean> } = {};
const missingMessageErrors: I18NMessageDictMap<Error> = {};
const missingSubstitutionErrors: {
  [localeCode: string]: I18NMessageDictMap<boolean>;
} = {};
```

Reply:
olala this is super helpful and i can't agree more with your point to have decent types than `any`. and thanks for showing this example, super clear ❤️